### PR TITLE
feat: add lambda executor

### PR DESCRIPTION
**Pull request by collindutter**

### CODE REVIEW

**Comment by collindutter** [comment]
Relevant to https://github.com/griptape-ai/griptape/issues/43

File: griptape/executors/lambda_executor.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            cwd=self.tool_dir(tool),
+            stdout=None if self.verbose else subprocess.DEVNULL,
+            stderr=None if self.verbose else subprocess.DEVNULL
+        )
+
+    def run_subprocess(self, env: dict[str, str], tool_activity: callable, value: Optional[dict]) -> subprocess.CompletedProcess:
+        tool = tool_activity.__self__
+        tool_name = tool.class_name
+        input_value = value if value else ""
+        deps_dir = f'/tmp/deps/{tool_name}/' if self.install_dependencies else '/var/task/'
```
Comment (vasinov) [inline griptape/executors/lambda_executor.py:62]: Can we move the `/tmp/deps` and `/var/task` into a class-level const?

File: griptape/executors/lambda_executor.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+        command = [
+            "python",
+            "-c",
+            f'from sys import path; path.insert(1, "{deps_dir}"); from tool import {tool_name}; print({tool_name}().{tool_activity.__name__}({input_value}))'
```
Comment (vasinov) [inline griptape/executors/lambda_executor.py:67]: I think those are starting to get long. We should just use string joins. Can you break this up into a list, join it with `str.join(";", <list name>)`, and apply the same pattern to other executors?

**Review by vasinov — CHANGES_REQUESTED** [review]
So, overall, this would only work if the whole structure is executed inside of a lambda function. It doesn't seem like we need a standalone executor to achieve whatever this new class is doing. All of those changes can be applied to the `LocalExecutor` to make it more versatile.

I think we should still consider building a standalone `AwsLambdaExecutor` that runs tools inside of lambda functions separately from where structures are executed. Basically, the same idea as the `DockerExecutor`.

**Pull request closed without merging**